src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -55,7 +55,15 @@ export function createEuiListItem({
   navigateToApp,
   dataTestSubj,
 }: Props) {
-  const { legacy, active, id, title, disabled, euiIconType, icon, tooltip, href } = link;
+  const { legacy, active, id, title, disabled, euiIconType, icon, tooltip } = link;
+  let { href } = link;
+
+  if (legacy) {
+    const legacyHref = link.url && !active ? link.url : link.baseUrl;
+    if (href !== legacyHref) {
+      href = legacyHref;
+    }
```

Comment:
nit: probably don't need this check
```suggestion
    href = link.url && !active ? link.url : link.baseUrl;
```